### ISSUE TITLE
sagittarius--socket.so not found by sagittarius--tls-socket.so

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#276]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/271) by Anonymous.**

----------------------------------------

Hi

Linuxbrew maintainer here. We are shipping sagittarius-scheme through our package manager an we noticed the following issue:

```
ldd /home/linuxbrew/.linuxbrew/Cellar/sagittarius-scheme/0.9.6_1/lib/sagittarius/0.9.6/x86_64-pc-linux/sagittarius--tls-socket.so
	linux-vdso.so.1 =>  (0x00007fffbed3d000)
	sagittarius--socket.so => not found
	libssl.so.1.0.0 => /home/linuxbrew/.linuxbrew/lib/libssl.so.1.0.0 (0x00007fa8b5296000)
	libcrypto.so.1.0.0 => /home/linuxbrew/.linuxbrew/lib/libcrypto.so.1.0.0 (0x00007fa8b4e39000)
	libsagittarius.so.0.9.6 => /home/linuxbrew/.linuxbrew/lib/libsagittarius.so.0.9.6 (0x00007fa8b4a2d000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fa8b4810000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fa8b460c000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fa8b4404000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fa8b40fb000)
	libgc.so.1 => /home/linuxbrew/.linuxbrew/lib/libgc.so.1 (0x00007fa8b3e7f000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fa8b3ab5000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fa8b5509000)
```

It looks like the linking failed and the sagittarius--socket.so can't be found. The file does exist, in the same folder:
```
ls /home/linuxbrew/.linuxbrew/Cellar/sagittarius-scheme/0.9.6_1/lib/sagittarius/0.9.6/x86_64-pc-linux/sagittarius--socket.so
```

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Adding RPATH handling (Fixes #271)

**Comment by ktakashi** [comment]
**Original comment by Michka Popoff (Bitbucket: [iMichka](https://bitbucket.org/iMichka), GitHub: [iMichka](https://github.com/iMichka)).**

----------------------------------------

Thank you, fix worked!
